components/script/dom/offscreencanvas.rs

```diff
@@ -0,0 +1,156 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+use dom::bindings::codegen::Bindings::OffscreenCanvasBinding;
+use dom::bindings::error::{Error, Fallible, report_pending_exception};
+use dom::bindings::codegen::Bindings::OffscreenCanvasBinding::OffscreenCanvasMethods;
+use dom::webglrenderingcontext::{LayoutCanvasWebGLRenderingContextHelpers, WebGLRenderingContext};
+use dom::offscreencanvasrenderingcontext2d::{OffscreenCanvasRenderingContext2D, LayoutOffscreenCanvasRenderingContext2DHelpers};
+use dom::htmlelement::HTMLElement;
+use dom::globalscope::GlobalScope;
+use dom::element::{Element, RawLayoutElementHelpers};
+use dom::bindings::cell::DomRefCell;
+use dom::bindings::reflector::{Reflector, reflect_dom_object};
+use dom::bindings::root::{Dom, DomRoot, LayoutDom};
+use dom::bindings::inheritance::Castable;
+use script_layout_interface::{HTMLCanvasData, HTMLCanvasDataSource};
+use dom::node::{Node, window_from_node};
+use html5ever::{LocalName, Prefix};
+use dom::bindings::root::Root;
+use dom::document::Document;
+use style::attr::{AttrValue, LengthOrPercentageOrAuto};
+
+
+use dom_struct::dom_struct;
+
+const DEFAULT_WIDTH: u32 = 300;
+const DEFAULT_HEIGHT: u32 = 150;
+
+#[must_root]
+#[derive(Clone, JSTraceable, MallocSizeOf)]
+pub enum CanvasContext {
+    Context2d(Dom<OffscreenCanvasRenderingContext2D>),
+    WebGL(Dom<WebGLRenderingContext>),
+}
+
+pub trait LayoutOffscreenCanvasHelpers {
+    fn data(&self) -> HTMLCanvasData;
+//	    fn get_width(&self) -> LengthOrPercentageOrAuto;
+//	    fn get_height(&self) -> LengthOrPercentageOrAuto;
+
+	}
+
+	#[dom_struct]
+	pub struct OffscreenCanvas {
+	    htmlelement: HTMLElement,
```

Comment:
This should be `reflector_: Reflector` instead, since an OffscreenCanvas is not an HTML element.